src/ibex_bluesky_core/callbacks/plotting.py

```diff
@@ -74,3 +86,49 @@ def start(self, doc: RunStart) -> None:
         """Process an start document (delegate to superclass, then show the plot)."""
         super().start(doc)
         show_plot()
+
+
+class PlotPNGSaver(QtAwareCallback):
+    """Save plots to PNG files on a run end."""
+
+    def __init__(
+        self,
+        x: str,
+        y: str,
+        ax: Axes | None,
+        postfix: str,
+        output_dir: str | os.PathLike[str] | None,
+    ) -> None:
+        """Initialise the PlotPNGSaver callback.
+
+        Args:
+            x: The name of the signal for x.
+            y: The name of the signal for y.
+            ax: The subplot to save to a file.
+            postfix: The file postfix.
+            output_dir: The output directory for PNGs.
+
+        """
+        super().__init__()
+        if ax is None:
+            raise ValueError("ax is None")
+        self.x = x
+        self.y = y
+        self.ax = ax
+        self.postfix = postfix
+        self.output_dir = Path(output_dir or get_default_output_path())
+
+    def stop(self, doc: RunStop) -> None:
+        """Write the current plot to a PNG file.
+
+        Args:
+            doc: The stop document.
+
+        """
+        rb_num = _get_rb_num(doc)
```

Comment:
Does this actually work - there is no rb_num in a stop document?

It's in the _start_ document, so you'll have to handle a start doc too and stash the RB number from it so that you can use it in this cb.